components/content_settings/core/browser/brave_content_settings_utils.cc

```diff
@@ -21,6 +23,32 @@ const std::vector<std::string> kShieldsResourceIDs {
     brave_shields::kReferrers,
     brave_shields::kCookies };
 
+bool CanPatternBeConvertedToWildcardSchemeAndPort(
+    const ContentSettingsPattern& pattern) {
+  // 1. Wildcard is alerady in the desired state.
+  // 2. Our firstParty placeholder shouldn't be converted.
+  // 3. Patterns that have file:// scheme.
+  // 4. We only want to convert patterns that have a specific host, so something
+  // like "http://*:80/*" should be left alone.
+  if (pattern == ContentSettingsPattern::Wildcard() ||
+      pattern == ContentSettingsPattern::FromString("https://firstParty/*") ||
+      pattern.GetScheme() == ContentSettingsPattern::SCHEME_FILE ||
+      pattern.MatchesAllHosts())
+    return false;
+  // Check for the case when the scheme is wildcard, but the port isn't.
+  if (pattern.GetScheme() == ContentSettingsPattern::SCHEME_WILDCARD) {
+    GURL check_for_port_url("http://" + pattern.ToString());
+    return check_for_port_url.has_port();
+  }
+  GURL url(pattern.ToString());
+  if (url.is_empty())
```

Comment:
mb we should check `is_valid()`

Reply:
Fixed in bdc2851b5386b1ae42a134550a217edb6267ef57